pallets/parachain-staking/src/lib.rs

```diff
@@ -2284,6 +2324,20 @@ pub mod pallet {
 			// prepare unstaking of collator candidate
 			Self::prep_unstake_exit_queue(&state.id, state.stake);
 
+			// disable validator for next session if they were in the set of validators
+			if let Some(index) = pallet_session::Pallet::<T>::validators()
+				.into_iter()
+				.enumerate()
+				.find_map(|(i, id): (usize, T::ValidatorId)| {
+					if <T as pallet_session::Config>::ValidatorIdOf::convert(collator.clone()) == Some(id) {
+						Some(i)
+					} else {
+						None
+					}
+				}) {
+				pallet_session::Pallet::<T>::disable_index(index);
```

Comment:
Unfortunately this does not seem to prevent this collator from being able to produce a block. I tried it out locally twice, and the kicked collator still produced blocks. 

I asked in Element and did not receive any answer yet. There is an [open issue for this with Babe instead of AuRa](https://github.com/paritytech/substrate/issues/8004).

Reply:
maybe add a `FIXME: ...` in the code?